script/bootstrap

```diff
@@ -2,45 +2,61 @@
 
 'use strict'
 
-const path = require('path')
-const CONFIG = require('./config')
-const childProcess = require('child_process')
-const cleanDependencies = require('./lib/clean-dependencies')
-const deleteMsbuildFromPath = require('./lib/delete-msbuild-from-path')
-const dependenciesFingerprint = require('./lib/dependencies-fingerprint')
-const installApm = require('./lib/install-apm')
-const runApmInstall = require('./lib/run-apm-install')
-const installScriptDependencies = require('./lib/install-script-dependencies')
-const verifyMachineRequirements = require('./lib/verify-machine-requirements')
-
-process.on('unhandledRejection', function (e) {
-  console.error(e.stack || e)
-  process.exit(1)
-})
-
-// We can't use yargs until installScriptDependencies() is executed, so...
-let ci = process.argv.indexOf('--ci') !== -1
-
-if (!ci && process.env.CI === 'true' && process.argv.indexOf('--no-ci') === -1) {
-  console.log('Automatically enabling --ci because CI is set in the environment')
-  ci = true
+async function bootstrap() {
+  const path = require('path')
+  const CONFIG = require('./config')
+  const childProcess = require('child_process')
+  const cleanDependencies = require('./lib/clean-dependencies')
+  const deleteMsbuildFromPath = require('./lib/delete-msbuild-from-path')
+  const dependenciesFingerprint = require('./lib/dependencies-fingerprint')
+  const installApm = require('./lib/install-apm')
+  const runApmInstall = require('./lib/run-apm-install')
+  const installScriptDependencies = require('./lib/install-script-dependencies')
+  const verifyMachineRequirements = require('./lib/verify-machine-requirements')
+
+  process.on('unhandledRejection', function (e) {
+    console.error(e.stack || e)
+    process.exit(1)
+  })
+
+  // We can't use yargs until installScriptDependencies() is executed, so...
+  let ci = process.argv.indexOf('--ci') !== -1
+
+  if (!ci && process.env.CI === 'true' && process.argv.indexOf('--no-ci') === -1) {
+    console.log('Automatically enabling --ci because CI is set in the environment')
+    ci = true
+  }
+
+  verifyMachineRequirements(ci)
+
+  if (dependenciesFingerprint.isOutdated()) {
+    cleanDependencies()
+  }
+
+  if (process.platform === 'win32') deleteMsbuildFromPath()
+
+  process.env.JOBS = "max";
+
+  await Promise.all([
+    installScriptDependencies(ci),
+    installApm(ci, true)
+      .then(() => {
+        childProcess.execFileSync(
+          CONFIG.getApmBinPath(),
+          ['--version'],
+          { stdio: 'inherit' }
+        );
+      })
+      .then(() => runApmInstall(CONFIG.repositoryRootPath, ci, undefined, true)),
+  ]);
```

Comment:
Main change (git does not like wrapping sections of code into a function it seems).

Adds the JOBS=max environment config for node-gyp.

Makes the script dependencies install concurrently with apm + apm install. The other parts are too quick to be worth making concurrent (except perhaps the `cleanDependencies` function; the effect can only be seen locally though, so will need to run more tests with that and maybe make a followup PR).